apps/saleor-app-checkout/backend/payments/providers/adyen/createPayment.ts

```diff
@@ -56,17 +61,84 @@ const createPaymentLink = (
           street: order.shippingAddress.streetAddress1,
           houseNumberOrName: order.shippingAddress.streetAddress2,
           postalCode: order.shippingAddress.postalCode,
-          stateOrProvince: order.shippingAddress.countryArea,
+          // stateOrProvince: order.shippingAddress.countryArea,
+          stateOrProvince: "New York",
         }
       : undefined,
+  };
+};
+
+export const createAdyenCheckoutPaymentLinks = async ({
+  order,
+  redirectUrl,
+}: CreatePaymentData) => {
+  const { config, checkout } = await getAdyenClient();
+  invariant(config.merchantAccount, "Missing merchant account configuration");
+
+  return checkout.paymentLinks(
+    orderToAdyenRequest({
+      order,
+      merchantAccount: config.merchantAccount,
+      returnUrl: formatRedirectUrl(redirectUrl, order.id),
+    })
+  );
+};
+
+export const createAdyenCheckoutSession = async ({
+  currency,
+  totalAmount,
+  checkoutId,
+  redirectUrl,
+}: {
+  currency: string;
+  totalAmount: number;
+  checkoutId: string;
+  redirectUrl: string;
+}) => {
+  const { config, checkout } = await getAdyenClient();
+  invariant(config.merchantAccount, "Missing merchant account configuration");
```

Comment:
Isn't something that could/should be placed in `getAdyenClient` ? You're duplicate the line `invariant(config.merchantAccount...` for all calls of this method  in this file